piop/src/sumcheck/tests/utils.rs

```diff
@@ -0,0 +1,110 @@
+use ark_std::{end_timer, start_timer, vec::Vec};
+use crypto_bigint::Random;
+use crypto_primitives::PrimeField;
+use rand::{Rng, RngCore};
+use std::sync::Arc;
+use zinc_poly::{mle::DenseMultilinearExtension, utils::ArithErrors};
+
+#[allow(clippy::arithmetic_side_effects)]
+#[allow(clippy::type_complexity)]
+pub(crate) fn rand_poly<F: PrimeField + Random, Rn: RngCore>(
+    nv: usize,
+    num_multiplicands_range: (usize, usize),
+    num_products: usize,
+    rng: &mut Rn,
+    config: &F::Config,
+) -> Result<
+    (
+        (Vec<DenseMultilinearExtension<F>>, usize),
+        Vec<(F, Vec<usize>)>,
+        F,
+    ),
+    ArithErrors,
+> {
+    let mut sum = F::zero_with_cfg(config);
+    let mut mles = vec![];
+    let mut products = Vec::with_capacity(num_products);
+    let mut degree = 0;
+    let mut current_mle_index = 0;
+    for _ in 0..num_products {
+        let num_multiplicands =
+            rng.random_range(num_multiplicands_range.0..num_multiplicands_range.1);
```

Comment:
With the change above this would become:
```suggestion
            rng.random_range(num_multiplicands_range.clone());
```